### DISCUSSION TITLE
Serverless deployment

### DISCUSSION THREAD

**Discussion by Sytten** [ory/kratos#1180] [Q&A] [answered]
I really like the idea of kratos (and other ory products) but all my projects are based on serverless architecture. I was wondering if it was possible to run it on say lambda or cloud run (basically frozen execution between requests). And if so if there are some limitations I should be aware of.

**Comment by zepatrik** [comment, marked as answer]
Kratos has no native serverless support.
I'd say that the request handlers work fine in a serverless deployment. You would have to figure out the database connection and some of the global state that is available through the central [registry](https://github.com/ory/kratos/blob/15ade1c5a74485fa07b41e5da00dc7229207ec88/driver/registry_default.go) as well as background tasks such as the email courier worker (#1033). Maybe you would be interested in a hosted Kratos managed by Ory instead? It is easier than serverless deployment :wink:
If so, PM @jaredpreston on our slack or drop us an email at hi@ory.sh :wave:

**Reply by Sytten** [reply]
Good to know. Yeah usually the annoying part is the background worker, the rest is fine (db connections can cause issues if the pool isn't coded to support closed sockets).
Maybe a managed service would be better yes, depending on the pricing (serverless is very cheap).

**Reply by zepatrik** [reply]
We're launching a closed beta very soon, reach out to be part of that 😉

**Reply by vinckr** [reply]
Hey @Sytten,
check out the updated website for our early access : https://www.ory.sh/ .
